common/core/gpudevice.cpp

```diff
@@ -201,6 +201,8 @@ void GpuDevice::DisplayManager::HotPlugEventHandler() {
           "UpdateDisplayState.");
       UpdateDisplayState();
     }
+    /* Adding sleep to avoid busy looping before a retry */
+    usleep(1000);
```

Comment:
> If you add lseek here I think you are just adding one more system call.

IMO it would be more important to make the existing code more robust than worry about performance of a single system call. Adding sleeps to the code indicates that there is problem with the whole approach here.